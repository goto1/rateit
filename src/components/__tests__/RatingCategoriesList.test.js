import React from "react";
import RatingCategoriesList from "../RatingCategoriesList";
import renderer from "react-test-renderer";
import { shallow, render } from "enzyme";

const props = {
  ratings: [
    {
      description: "Attends Group Meetings",
      rating: 3
    },
    {
      description: "Contributes to Discussions",
      rating: 5
    }
  ],
  hrColors: {
    colorOne: "#FFF",
    colorTwo: "#000"
  }
};

describe("RatingCategoriesList component", () => {
  it("should render correctly", () => {
    const tree = renderer.create(<RatingCategoriesList {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it("should render the right amount of rating categories", () => {
    const wrapper = shallow(<RatingCategoriesList {...props} />);
    const actual = wrapper.find("div").length;
    const expected = 2;
    expect(actual).toBe(expected);
  });

  it("should render the correct rating category", () => {
    const wrapper = render(<RatingCategoriesList {...props} />);
    const actual = wrapper.html().includes("Attends Group Meetings");
    const expected = true;
    expect(actual).toBe(expected);
  });
});
