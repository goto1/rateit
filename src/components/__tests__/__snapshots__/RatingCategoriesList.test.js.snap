// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingCategoriesList component should render correctly 1`] = `
.ecEnce {
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
}

.eYvYrI {
  font-weight: 500;
  color: #DEB21C;
}

.fxvkdv {
  font-weight: 500;
  color: #A1BA37;
}

.ixBfIh {
  border: 0px;
  height: 1px;
  width: 70%;
  margin: 1px auto;
  background-image: linear-gradient(    to left,    #FFF,    #000,    #FFF  );
}

.dhCMem {
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  padding-top: 4px;
  color: #000;
}

.gcHgtx {
  -webkit-flex-basis: 22px;
  -ms-flex-basis: 22px;
  flex-basis: 22px;
  font-size: 19px;
  padding-top: 4px;
}

.gbCgzD {
  font-size: 14px;
}

.gbCgzD span {
  font-size: 17px;
}

.kdJdmo {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 1px;
}

.geAHSS {
  margin-top: 7.5px;
}

<div
  className="sc-kgoBCf geAHSS"
>
  .ecEnce {
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
}

.eYvYrI {
  font-weight: 500;
  color: #DEB21C;
}

.ixBfIh {
  border: 0px;
  height: 1px;
  width: 70%;
  margin: 1px auto;
  background-image: linear-gradient(    to left,    #FFF,    #000,    #FFF  );
}

.dhCMem {
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  padding-top: 4px;
  color: #000;
}

.gcHgtx {
  -webkit-flex-basis: 22px;
  -ms-flex-basis: 22px;
  flex-basis: 22px;
  font-size: 19px;
  padding-top: 4px;
}

.gbCgzD {
  font-size: 14px;
}

.gbCgzD span {
  font-size: 17px;
}

.kdJdmo {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 1px;
}

<div>
    .ecEnce {
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
}

.eYvYrI {
  font-weight: 500;
  color: #DEB21C;
}

.dhCMem {
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  padding-top: 4px;
  color: #000;
}

.gcHgtx {
  -webkit-flex-basis: 22px;
  -ms-flex-basis: 22px;
  flex-basis: 22px;
  font-size: 19px;
  padding-top: 4px;
}

.gbCgzD {
  font-size: 14px;
}

.gbCgzD span {
  font-size: 17px;
}

.kdJdmo {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 1px;
}

<div
      className="sc-chPdSV kdJdmo"
    >
      .gcHgtx {
  -webkit-flex-basis: 22px;
  -ms-flex-basis: 22px;
  flex-basis: 22px;
  font-size: 19px;
  padding-top: 4px;
}

<i
        className="material-icons icon sc-cSHVUG gcHgtx"
      >
        forward
      </i>
      .dhCMem {
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  padding-top: 4px;
  color: #000;
}

<div
        className="sc-jzJRlG dhCMem"
      >
        Attends Group Meetings
      </div>
      .ecEnce {
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
}

.eYvYrI {
  font-weight: 500;
  color: #DEB21C;
}

.gbCgzD {
  font-size: 14px;
}

.gbCgzD span {
  font-size: 17px;
}

<div
        className="sc-kAzzGY gbCgzD sc-bdVaJa ecEnce"
      >
        .eYvYrI {
  font-weight: 500;
  color: #DEB21C;
}

<span
          className="sc-bwzfXH eYvYrI"
          color="#DEB21C"
        >
          3
        </span>
        /5
      </div>
    </div>
    .ixBfIh {
  border: 0px;
  height: 1px;
  width: 70%;
  margin: 1px auto;
  background-image: linear-gradient(    to left,    #FFF,    #000,    #FFF  );
}

<hr
      className="sc-htpNat ixBfIh"
      width="70%"
    />
  </div>
  .ecEnce {
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
}

.fxvkdv {
  font-weight: 500;
  color: #A1BA37;
}

.dhCMem {
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  padding-top: 4px;
  color: #000;
}

.gcHgtx {
  -webkit-flex-basis: 22px;
  -ms-flex-basis: 22px;
  flex-basis: 22px;
  font-size: 19px;
  padding-top: 4px;
}

.gbCgzD {
  font-size: 14px;
}

.gbCgzD span {
  font-size: 17px;
}

.kdJdmo {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 1px;
}

<div>
    .ecEnce {
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
}

.fxvkdv {
  font-weight: 500;
  color: #A1BA37;
}

.dhCMem {
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  padding-top: 4px;
  color: #000;
}

.gcHgtx {
  -webkit-flex-basis: 22px;
  -ms-flex-basis: 22px;
  flex-basis: 22px;
  font-size: 19px;
  padding-top: 4px;
}

.gbCgzD {
  font-size: 14px;
}

.gbCgzD span {
  font-size: 17px;
}

.kdJdmo {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 1px;
}

<div
      className="sc-chPdSV kdJdmo"
    >
      .gcHgtx {
  -webkit-flex-basis: 22px;
  -ms-flex-basis: 22px;
  flex-basis: 22px;
  font-size: 19px;
  padding-top: 4px;
}

<i
        className="material-icons icon sc-cSHVUG gcHgtx"
      >
        forward
      </i>
      .dhCMem {
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  padding-top: 4px;
  color: #000;
}

<div
        className="sc-jzJRlG dhCMem"
      >
        Contributes to Discussions
      </div>
      .ecEnce {
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
}

.fxvkdv {
  font-weight: 500;
  color: #A1BA37;
}

.gbCgzD {
  font-size: 14px;
}

.gbCgzD span {
  font-size: 17px;
}

<div
        className="sc-kAzzGY gbCgzD sc-bdVaJa ecEnce"
      >
        .fxvkdv {
  font-weight: 500;
  color: #A1BA37;
}

<span
          className="sc-bwzfXH fxvkdv"
          color="#A1BA37"
        >
          5
        </span>
        /5
      </div>
    </div>
  </div>
</div>
`;
